### PR TITLE
Fix drag-to-zoom (rebased)

### PR DESCRIPTION
**Pull request by rasmusfaber**
This fixes "Drag-zoom does not zoom to marked area #154"

### CODE REVIEW

**Comment by rasmusfaber** [comment]
This pull request also contains a simplification of zoomTimeScale(): there is no need to convert back and forth between values and pixels, but if you want, I can remove that part to minimize the impact.

**Comment by JanOonk** [comment]
I can confirm this fix works! Thanks.

**Comment by benmccann** [comment]
@rasmusfaber this PR will need to be rebased

**Comment by rasmusfaber** [comment]
It has now been rebased.

File: src/chart.zoom.js [hunk 1]
```diff
@@ -370,19 +364,21 @@ var zoomPlugin = {
...
 					var beginPoint = chartInstance.zoom._dragZoomStart;
 					var offsetX = beginPoint.target.getBoundingClientRect().left;
-					var startX = Math.min(beginPoint.clientX, event.clientX) - offsetX;
-					var endX = Math.max(beginPoint.clientX, event.clientX) - offsetX;
-					var dragDistance = endX - startX;
-					var chartDistance = chartArea.right - chartArea.left;
-					var zoom = 1 + ((chartDistance - dragDistance) / chartDistance );
+					var startX = Math.max(Math.min(beginPoint.clientX, event.clientX) - offsetX,
+                                chartArea.left);
+                    var endX = Math.min(Math.max(beginPoint.clientX, event.clientX) - offsetX, chartArea.right);
```
Comment (benmccann) [inline src/chart.zoom.js]: please use tabs instead of spaces for consistency

File: src/chart.zoom.js [hunk 2]
```diff
@@ -108,29 +108,23 @@ function zoomIndexScale(scale, zoom, center, zoomOptions) {
...
+	var max_percent = 1 - min_percent;
+	
 	var minDelta = newDiff * min_percent;
 	var maxDelta = newDiff * max_percent;
-
-	var newMin = scale.getValueForPixel(scale.getPixelForValue(scale.min) + minDelta);
-	var newMax = scale.getValueForPixel(scale.getPixelForValue(scale.max) - maxDelta);
-
-	var diffMinMax = newMax.diff(newMin);
+
```
Comment (benmccann) [inline src/chart.zoom.js]: please remove the trailing whitespace here and on the lines below

**Comment by rasmusfaber** [comment]
Sorry about the formatting, it should be fixed now.

**Comment by benmccann** [comment]
Thanks for putting together a fix for this. I believe this issue has now been fixed by https://github.com/chartjs/chartjs-plugin-zoom/pull/161, so I'm going to close this PR. Please let me know if you still think there's any issue remaining

**Pull request closed without merging**